egs/tedlium/s5/local/run_cleanup_segmentation.sh

```diff
@@ -0,0 +1,111 @@
+#!/bin/bash
+
+# Copyright 2016  Vimal Manohar
+# Apache 2.0
+
+# This script demonstrates how to re-segment training data selecting only the 
+# "good" audio that matches the transcripts.
+# The basic idea is to decode with an existing in-domain acoustic model, and a
+# biased language model built from the reference, and then work out the
+# segmentation from a ctm like file.
+
+set -e 
+set -o pipefail 
+set -u
+
+stage=0
+cleanup_stage=0
+
+cleanup_affix=cleaned
+nj=100
+
+pad_length=2
+max_silence_length=50
+max_incorrect_words=0
+min_correct_frames=0
+max_segment_wer=20000
+min_wer_for_splitting=10
+
+ngram_order=2
+
+interpolation_weights=0:0.1:0.9
+top_words_interpolation_weight=0.1
+
+group_utt_words=
+
+. ./path.sh
+. ./cmd.sh
+. utils/parse_options.sh 
+
+nj=405
+
+# Set the variables. These are based on variables set by run_ivector_common.sh
+gmm_dir=exp/tri3
+
+bad_utts_dir=${gmm_dir}_train_split_bad_utts 
+
+lm_affix="o$ngram_order"
+if [ $ngram_order -eq 1 ]; then
+  lm_affix=${lm_affix}_top$top_n_words
+fi
+
+bad_utts_dir=${bad_utts_dir}_${lm_affix}
+
+if [ ! -z $group_utt_words ]; then
+  bad_utts_dir=${bad_utts_dir}_group$group_utt_words
+fi
+
+if [ $stage -le 1 ]; then
+  steps/cleanup/do_cleanup_segmentation.sh \
+    --cmd "$train_cmd" --nj $nj \
+    --pad-length $pad_length \
+    --max-silence-length $max_silence_length \
+    --max-incorrect-words $max_incorrect_words \
+    --max-segment-wer $max_segment_wer \
+    --min-correct-frames $min_correct_frames \
```

Comment:
I think you have too many options here.  If you don't think options will need to be changed per corpus, you could remove most of them in these example scripts.  It makes them very cluttered.  In your private branch you can keep them if you're still tuning.